hack/verify-generated-docs.sh

```diff
@@ -28,57 +28,31 @@ kube::golang::setup_env
     cmd/genkubedocs \
     cmd/genman \
     cmd/genyaml \
-    cmd/mungedocs
-
-# Find binary
-mungedocs=$(kube::util::find-binary "mungedocs")
-
-DOCROOT="${KUBE_ROOT}/docs/"
-EXAMPLEROOT="${KUBE_ROOT}/examples/"
-
-# mungedocs --verify can (and should) be run on the real docs, otherwise their
-# links will be distorted. --verify means that it will not make changes.
-# --verbose gives us output we can use for a diff.
-"${mungedocs}" "--verify=true" "--verbose=true" "--upstream=${KUBE_GIT_UPSTREAM}" "--root-dir=${DOCROOT}" && ret=0 || ret=$?
-if [[ $ret -eq 1 ]]; then
-  echo "${DOCROOT} is out of date. Please run hack/update-generated-docs.sh"
-  exit 1
-fi
-if [[ $ret -gt 1 ]]; then
-  echo "Error running mungedocs"
-  exit 1
-fi
-
-"${mungedocs}" "--verify=true" "--verbose=true" "--upstream=${KUBE_GIT_UPSTREAM}" "--root-dir=${EXAMPLEROOT}" && ret=0 || ret=$?
-if [[ $ret -eq 1 ]]; then
-  echo "${EXAMPLEROOT} is out of date. Please run hack/update-generated-docs.sh"
-  exit 1
-fi
-if [[ $ret -gt 1 ]]; then
-  echo "Error running mungedocs"
-  exit 1
-fi
+    federation/cmd/genfeddocs
 
 kube::util::ensure-temp-dir
 
 kube::util::gen-docs "${KUBE_TEMP}"
-diff -Naup "${KUBE_TEMP}/.generated_docs" "${KUBE_ROOT}/.generated_docs" || ret=1 || true
-while read file; do
-  diff -Naup "${KUBE_TEMP}/${file}" "${KUBE_ROOT}/${file}" || ret=1 || true
-done <"${KUBE_TEMP}/.generated_docs"
 
-needsanalytics=($(kube::util::gen-analytics "${KUBE_ROOT}" 1))
-if [[ ${#needsanalytics[@]} -ne 0 ]]; then
-  echo -e "Some md files are missing ga-beacon analytics link:"
-  printf '%s\n' "${needsanalytics[@]}"
-  ret=1
-fi
-if [[ $ret -eq 0 ]]
-then
-  echo "Generated docs are up to date."
-else
-  echo "Generated docs are out of date. Please run hack/update-generated-docs.sh"
+# Verify the list matches the expected list (diff should be empty)
+if [[ "$(diff ${KUBE_ROOT}/.generated_docs ${KUBE_TEMP}/.generated_docs)" != "" ]]; then
```

Comment:
Preference, but could also be written as:
if diff -q file1 file2 >/dev/null 2>&1